### ISSUE TITLE
Doesn't seem to work with Python 3 (3.2, specifically)

### ISSUE THREAD

**Issue by liftoff** [steeve/python-lz4#7]
Just tried installing python-lz4 on Ubuntu 12.10 using python3 (which is python 3.2.3).  When trying to import the lz4 module I get the following error:

```
>>> import lz4
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
ImportError: /usr/local/lib/python3.2/dist-packages/lz4-0.6.0-py3.2-linux-x86_64.egg/lz4.cpython-32mu.so: undefined symbol: PyString_FromStringAndSize
```

I assume this is because python-lz4 does not support Python 3 yet?  Here's the output of `sudo python3 setup.py install`:

```
running install
Checking .pth file support in /usr/local/lib/python3.2/dist-packages/
/usr/bin/python3 -E -c pass
TEST PASSED: /usr/local/lib/python3.2/dist-packages/ appears to support .pth files
running bdist_egg
running egg_info
creating src/lz4.egg-info
writing src/lz4.egg-info/PKG-INFO
writing top-level names to src/lz4.egg-info/top_level.txt
writing dependency_links to src/lz4.egg-info/dependency_links.txt
writing manifest file 'src/lz4.egg-info/SOURCES.txt'
reading manifest file 'src/lz4.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/lz4.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_ext
building 'lz4' extension
creating build
creating build/temp.linux-x86_64-3.2
creating build/temp.linux-x86_64-3.2/src
gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -I/usr/include/python3.2mu -c src/lz4.c -o build/temp.linux-x86_64-3.2/src/lz4.o -O4
gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -I/usr/include/python3.2mu -c src/lz4hc.c -o build/temp.linux-x86_64-3.2/src/lz4hc.o -O4
gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -I/usr/include/python3.2mu -c src/python-lz4.c -o build/temp.linux-x86_64-3.2/src/python-lz4.o -O4
In file included from /usr/include/python3.2mu/Python.h:8:0,
                from src/python-lz4.c:3:
/usr/include/python3.2mu/pyconfig.h:1182:0: warning: "_POSIX_C_SOURCE" redefined [enabled by default]
In file included from /usr/include/stdlib.h:25:0,
                from src/python-lz4.c:1:
/usr/include/features.h:215:0: note: this is the location of the previous definition
src/python-lz4.c: In function ‘compress_with’:
src/python-lz4.c:41:5: warning: implicit declaration of function ‘PyString_FromStringAndSize’ [-Wimplicit-function-declaration]
src/python-lz4.c:41:12: warning: assignment makes pointer from integer without a cast [enabled by default]
src/python-lz4.c:44:5: warning: implicit declaration of function ‘PyString_AS_STRING’ [-Wimplicit-function-declaration]
src/python-lz4.c:44:10: warning: assignment makes pointer from integer without a cast [enabled by default]
src/python-lz4.c:51:6: warning: implicit declaration of function ‘_PyString_Resize’ [-Wimplicit-function-declaration]
src/python-lz4.c: In function ‘py_lz4_uncompress’:
src/python-lz4.c:89:12: warning: assignment makes pointer from integer without a cast [enabled by default]
src/python-lz4.c:91:15: warning: initialization makes pointer from integer without a cast [enabled by default]
src/python-lz4.c: In function ‘initlz4’:
src/python-lz4.c:124:5: warning: implicit declaration of function ‘Py_InitModule’ [-Wimplicit-function-declaration]
src/python-lz4.c:124:7: warning: assignment makes pointer from integer without a cast [enabled by default]
src/python-lz4.c:126:9: warning: ‘return’ with no value, in function returning non-void [-Wreturn-type]
creating build/lib.linux-x86_64-3.2
gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-3.2/src/lz4.o build/temp.linux-x86_64-3.2/src/lz4hc.o build/temp.linux-x86_64-3.2/src/python-lz4.o -o build/lib.linux-x86_64-3.2/lz4.cpython-32mu.so
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
copying build/lib.linux-x86_64-3.2/lz4.cpython-32mu.so -> build/bdist.linux-x86_64/egg
creating stub loader for lz4.cpython-32mu.so
byte-compiling build/bdist.linux-x86_64/egg/lz4.py to lz4.cpython-32.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying src/lz4.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/lz4.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/lz4.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/lz4.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
__pycache__.lz4.cpython-32: module references __file__
creating dist
creating 'dist/lz4-0.6.0-py3.2-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing lz4-0.6.0-py3.2-linux-x86_64.egg
creating /usr/local/lib/python3.2/dist-packages/lz4-0.6.0-py3.2-linux-x86_64.egg
Extracting lz4-0.6.0-py3.2-linux-x86_64.egg to /usr/local/lib/python3.2/dist-packages
Adding lz4 0.6.0 to easy-install.pth file

Installed /usr/local/lib/python3.2/dist-packages/lz4-0.6.0-py3.2-linux-x86_64.egg
Processing dependencies for lz4==0.6.0
Finished processing dependencies for lz4==0.6.0
```

**Comment by steeve** [comment]
Thanks for you report!
I don't know the specs for Python 3 native modules, but from what I've gathered this is an issue for many native python modules.
I don't have the time to check up on that, but feel free to submit a PR if you have a fix!

http://docs.python.org/2/howto/cporting.html is a good starting point: it seems you just have to replace PyString_\* with PyBytes_\* on Python 3k.

**Comment by liftoff** [comment]
I am pretty good at porting pure Python 2 stuff to Python 3.  I've helped several projects make their code work Python 3.  Unfortunately I don't know a damned thing about writing Python modules in C so I am no help in that area :anguished:

I wish I did know how to port C modules to work with Python 3...  Then I could fix xpyb which I _really_ need to work with Python 3 soon (will be necessary in order to use Gate One's forthcoming web-based X11 feature--in Python 3 anyway).
